File: LeetCodeTasks/38_count-and-say/solutions/java/solution.java
class Solution {
    public String countAndSay(int n) {
        if (n == 1) return "1";
        String previous = countAndSay(n - 1);
        StringBuilder result = new StringBuilder();
        
        int count = 0;
        char currentChar = previous.charAt(0);
        
        for (int i = 0; i < previous.length(); i++) {
            if (previous.charAt(i) == currentChar) {
                count++;
            } else {
                result.append(count).append(currentChar);
                currentChar = previous.charAt(i);
                count = 1;
            }
        }
        
        result.append(count).append(currentChar);
        return result.toString();
    }
}

File: LeetCodeTasks/38_count-and-say/solutions/c/solution.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

char* countAndSay(int n) {
    if (n == 1) return strdup("1");

    char* previous = countAndSay(n - 1);
    int length = strlen(previous);
    int count = 1;
    char* result = malloc(length * 2 + 1); // Allocating max possible space needed
    int resultIndex = 0;

    for (int i = 1; i <= length; i++) {
        if (i < length && previous[i] == previous[i - 1]) {
            count++;
        } else {
            int addedLength = sprintf(result + resultIndex, "%d%c", count, previous[i - 1]);
            resultIndex += addedLength;
            count = 1;
        }
    }

    free(previous);
    return result;
}
